test/unified_malloc_framework/memoryPool.hpp

```diff
@@ -251,6 +252,36 @@ TEST_P(umfPoolTest, multiThreadedMallocFreeRandomSizes) {
     }
 }
 
+TEST(umfMemTest, outOfMem) {
```

Comment:
Right now, we are just verifying correctness our test helpers (we only use provider_mock_out_of_mem and proxy pool). We would like to use this outOfMem to test actual pool implementations (right now, DisjointPool). The best way would be to make this TEST parametrized (TEST_P macro, similarly as we have for all umfPoolTests) and instantiate it for DisjointPool and proxy_pool (see INSTANTIATE_TEST_SUITE_P macro, like here: https://github.com/oneapi-src/unified-runtime/blob/acca790cd6d12bd38b899c4320eb3e7aabf667ff/test/unified_malloc_framework/umf_pools/disjoint_pool.cpp#L72)